File: ec.go
package main

import (
    "fmt"
    "strings"
)

var SUBC = "spdfghijklmnoqrtuvwxyz"

type Subshell struct {
    N int
    C string
    MaxE int
}

func newSubshell(n int, c string) Subshell {
    var subshell Subshell
    
    subshell.N = n
    subshell.C = c
    subshell.MaxE = 2 * (2 * strings.Index(SUBC, c) + 1)
    
    return subshell
}

func (s *Subshell) Advance() {
    var nN int
    var nC string
    
    if s.C == "s" {
        nN = (s.N + 1) / 2 + 1;
        nC = string(SUBC[nN - (s.N % 2) - 1])
    } else {
        nN = s.N + 1
        nC = string(SUBC[strings.Index(SUBC, s.C) - 1])
    }
    
    s.N = nN
    s.C = nC
    s.MaxE = 2 * (2 * strings.Index(SUBC, nC) + 1)
}

func main() {
    var an int
    
    cSubshell := newSubshell(1, "s")
    
    fmt.Print("\nEnter the atomic number: ")
    fmt.Scan(&an)
    for an > 0 {
        fmt.Print(cSubshell.N, cSubshell.C)
        if an - cSubshell.MaxE > 0 {
            fmt.Print(cSubshell.MaxE)    
        } else {
            fmt.Print(an)
        }
        an -= cSubshell.MaxE
        cSubshell.Advance() 
        fmt.Print(" ")
    }
}

File: ec.js
var Subshell, ssn;

ssn = "spdfghijklmnoqrtuvwxyz";
Subshell = {
    n: 1,
    s: 's',
    maxE: 2,
    advance: function() {
        var nN, nS;
        
        if (this.s === 's') {
            nN = Math.ceil(this.n / 2) + 1;
            nS = ssn.charAt(nN - (this.n % 2) - 1);
        } else {
            nN = this.n + 1;
            nS = ssn.charAt(ssn.indexOf(this.s) - 1);
        }
        this.n = nN;
        this.s = nS;
        this.maxE = 2 * (2 * ssn.indexOf(nS) + 1);
    }
};

function getEC(aN) {
    var cSubshell, out;
    
    cSubshell = Object.create(Subshell);
    out = '';
    while (aN > 0) {
        out += cSubshell.n + cSubshell.s;
        if (aN - cSubshell.maxE > 0) {
            out += cSubshell.maxE;
        } else {
            out += aN;
        }
        aN -= cSubshell.maxE;
        cSubshell.advance();
        if (aN > 0) {
            out += ', ';
        }
    }
    return out;
}
